sdk/communication/AzureCommunicationCommon/Tests/CommunicationIdentifierHelperTests.swift

```diff
@@ -0,0 +1,87 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunicationCommon
+import XCTest
+
+class CommunicationIdentifierHelperTests: XCTestCase {
```

Comment:
I had thought we could have the exact same test cases for each language with the exact same inputs and outputs instead of having an own set per language. For creating identfier from raw id, getting raw if from identifier and roundtrip.

What do you think?
JS tests: https://github.com/Azure/azure-sdk-for-js/pull/21699/files#diff-1145d95d7955948a6d8eee7b691a65bc6c7aeb65f3a135c5eb47dcaa224eaae6

Reply:
It would definitely be easier to manage in the long run. I also used Dom's set of tests in C#.

Reply:
Updated the unit tests to align better with other platforms